packages/core/src/render3/instructions/element_validation.ts

```diff
@@ -0,0 +1,306 @@
+    // of HTMLUnknownElement in browsers that support web components.
+    const isUnknown =
+        // Note that we can't check for `typeof HTMLUnknownElement === 'function'`,
+        // because while most browsers return 'function', IE returns 'object'.
+        (typeof HTMLUnknownElement !== 'undefined' && HTMLUnknownElement &&
+         element instanceof HTMLUnknownElement) ||
+        (typeof customElements !== 'undefined' && tagName.indexOf('-') > -1 &&
+         !customElements.get(tagName));
+
+    if (isUnknown && !matchingSchemas(schemas, tagName)) {
+      const isHostStandalone = isHostComponentStandalone(lView);
+      const templateLocation = getTemplateLocationDetails(lView);
+      const schemas = `'${isHostStandalone ? '@Component' : '@NgModule'}.schemas'`;
+
+      let message = `'${tagName}' is not a known element${templateLocation}:\n`;
+      message += `1. If '${tagName}' is an Angular component, then verify that it is ${
+          isHostStandalone ? 'included in the \'@Component.imports\' of this component' :
+                             'a part of an @NgModule where this component is declared'}.\n`;
+      if (tagName && tagName.indexOf('-') > -1) {
+        message +=
+            `2. If '${tagName}' is a Web Component then add 'CUSTOM_ELEMENTS_SCHEMA' to the ${
+                schemas} of this component to suppress this message.`;
+      } else {
+        message +=
+            `2. To allow any element add 'NO_ERRORS_SCHEMA' to the ${schemas} of this component.`;
+      }
+      if (shouldThrowErrorOnUnknownElement) {
+        throw new RuntimeError(RuntimeErrorCode.UNKNOWN_ELEMENT, message);
+      } else {
+        console.error(formatRuntimeError(RuntimeErrorCode.UNKNOWN_ELEMENT, message));
+      }
+    }
+  }
+}
+
+/**
+ * Validates that the property of the element is known at runtime, if it's not the case
+ * produces an return false (otherwise returns true with no errors).
+ * This check is relevant for JIT-compiled components (for AOT-compiled
+ * ones this check happens at build time).
+ *
+ * The property is considered known if either:
+ * - it's a known property of the element
+ * - the element is allowed by one of the schemas
+ * - the property is used for animations
+ *
+ * @param element Element to validate
+ * @param tagName Name of the tag to check
+ * @param propName Name of the property to check
+ * @param schemas Array of schemas
+ * @param tNode `TNode` the element which presents the property
+ * @param lView An `LView` that represents a current component that is being rendered
+ */
+export function validateElementProperty(
+    element: RElement|RComment,
+    tagName: string|null,
+    propName: string,
+    schemas: SchemaMetadata[]|null,
+    tNode: TNode,
+    lView: LView,
+    ): boolean {
+  const propertyIsValid = isPropertyValid(element, tagName, propName, schemas);
+  if (!propertyIsValid) {
+    handleUnknownPropertyError(propName, tNode, lView);
+  }
+  return propertyIsValid;
```

Comment:
We can use a bit shorter name here (without loosing meaning and readability), for example:
```suggestion
  const isValid = isPropertyValid(element, tagName, propName, schemas);
  if (!isValid) {
    handleUnknownPropertyError(propName, tNode, lView);
  }
  return isValid;
```
another option:
```suggestion
  if (!isPropertyValid(element, tagName, propName, schemas)) {
    handleUnknownPropertyError(propName, tNode, lView);
    return false;
  }
  return true;
```

Reply:
I don't have any strong preference, feel free to choose the option that you see fit the best.